### PR TITLE
Reorganise Ruby tests

### PR DESCRIPTION
**Pull request by grodowski**
Reorganise to run minitests with `rake` and enable a few neat things:
- run a single test file, thanks to the absence of `main.rb`
- `ruby-lsp` goodness in Code: run/debug single test and test explorer

Screenshot 2024-12-06 at 11 24 40

FWIW I wanted to do the same for golang, but the current Go extension for Code requires `1.19+`, so a `go` update may be due in a separate PR. Any objections to do that?

### CODE REVIEW

**Comment by driv3r** [comment]
carve out logrus before updating golang please

File: .gitignore [hunk 1]
```diff
@@ -17,7 +17,6 @@ bin/
 .vagrant/
 build/
 .bundle/
-.vscode/
```
Comment (driv3r) [inline .gitignore:20]: why do we remove this from git ignore?

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "version": "0.2.0",
+    "configurations": [
+      {
+        "name": "Minitest - current file",
+        "type": "ruby_lsp",
+        "request": "launch",
+        "program": "rake test TEST=${file}"
+      }
+    ]
+  }
```
Comment (driv3r) [inline .vscode/launch.json:14]: please lets avoid any unnecessary IDE related staff out

**Review by driv3r — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 2 above)
Comment (grodowski) [reply .vscode/launch.json:14]: Good point, I added these only before finding out that adding `rake` might make sense, and now `ruby-lsp` runs tests just fine. Removing!

File: test/integration/trivial_test.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 require "test_helper"
 
-class TrivialIntegrationTests < GhostferryTestCase
+class TrivialIntegrationTest < GhostferryTestCase
```
Comment (grodowski) [inline test/integration/trivial_test.rb:3]: TIL: `ruby-lsp` doesn't recognise test classes with plural names

**Review by driv3r — APPROVED** [review]

**Pull request merged**